packages/backend-auth/API.md

```diff
@@ -86,6 +103,17 @@ export type OidcProviderFactoryProps = Omit<OidcProviderProps, 'clientId' | 'cli
     clientSecret: BackendSecret;
 };
 
+// @public (undocumented)
+export type ResourceAccessBuilder = {
+    to: (actions: UserpoolAction[]) => AccessDefinition;
+};
+
+// @public (undocumented)
+export type Role = 'users' | 'groups';

```

Comment:
Kind of hard to name it.

If we don't know perhaps we should inline this into `Partial<Record<'users' | 'groups', AccessDefinition>>` above. and punt the problem.

`users` and `groups` are `UserPoolEnitity` perhaps ?

Reply:
As per IAM terminology, these are either Principal or Identity https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-terms

Reply:
this is different than that. Here we are granted access to either the user actions (CreateUser, GetUser, etc) on the userpool or the group actions (CreateGroup, GetGroup, etc) on the userpool. This is not referring to an IAM User or IAM Group

Reply:
I thought it was CognitoUser or CognitoGroups (i.e. still Cognito Identity). In case this is referring to actions, perhaps

`export type UserPoolAction = 'userActions' | 'groupActions';`

Reply:
That's kinda where my head was at with the `ActionScope` suggestion. For example, a definition of
```ts
{
  users: allow.resource(myFunc).to(['read'])
}
```
Would mean "scope the read access of myFunc to UserPool users".

And
```ts
{
  groups: allow.resource(otherFunc).to(['read'])
}
```
Would mean "scope the read access of otherFunc to UserPool roles".

Reply:
`export type UserPoolAction = 'userActions' | 'groupActions';` this sounds good.

Reply:
`UserPoolActionScope` is fine too.

Reply:
I'm less worried about the type but the value here. I think this conveys a little more accurately
```ts
{
  userActions: [
    allow.resource(myFunc).to(['read'])
  ],
  groupActions: [
    allow.resource(otherFunc).to(['read'])
  ]
}
```

Reply:
I'm good with the above. Not sure we want to leak "UserPool" into the type name at this level of abstraction though. Maybe `AuthManagementActions = 'userActions' | 'groupActions'`?

Reply:
not relevant now, action scopes are removed